src/features/canvass/components/MarkerIcon.tsx

```diff
@@ -31,7 +32,14 @@ const MarkerIcon: FC<MarkerIconProps> = ({
         d="M10.5 0C4.695 0 0 4.695 0 10.5C0 18.375 10.5 30 10.5 30C10.5 30 21 18.375 21 10.5C21 4.695 16.305 0 10.5 0Z"
         fill={selected ? '#ED1C55' : 'white'}
       />
-
+      <clipPath id={visitsBackgroundKey}>
+        <rect height="30" width="21" x="0" y="0" />
+      </clipPath>
+      <path
+        clipPath={`url(#${visitsBackgroundKey})`}
+        d="M10.5 3C6 3 3 6.5 3 10.5C3 16 10.5 27 10.5 27C10.5 27 18 16 18 10.5C18 6.5 15 3 10.5 3Z"
+        fill={'white'}
+      />
```

Comment:
If I read this correctly, you have:

- Created a clipping path that is a rectangle big enough to fit the entire pin
- Created a white pin shape, with the rectangle above as the clipping path

Would it not be possible to just do a white pin shape without any clipping path.

```suggestion
      <path
        d="M10.5 3C6 3 3 6.5 3 10.5C3 16 10.5 27 10.5 27C10.5 27 18 16 18 10.5C18 6.5 15 3 10.5 3Z"
        fill={'white'}
      />
```

Reply:
I rewrote this significantly to try to simplify it. Now it draws three rects for the visualisation and clips them to the shape of the pin's interior space. I think this removes the need for the uniqueKey, so I took it out and it still seemed to work in my testing, but I may have overlooked why it's needed!

Reply:
Cool!

I believe the `uniqueKey` is needed because this SVG is rendered multiple times in the same HTML document, and each occurrence needs it's own set of IDs. Does that sound relevant?